qt/src/Kdjvu.hh

```diff
@@ -0,0 +1,301 @@
+         * \return a vector with the pages of the current document
+         */
+        const QVector<KDjVu::Page*> &pages() const;
+
+        /**
+         * Get the metadata for the specified \p key, or a null variant otherwise.
+         */
+        QVariant metaData( const QString & key ) const;
+
+        /**
+         * Get an XML document with the bookmarks of the current document (if any).
+         * The XML will look like this:
+         * \verbatim
+         * <!DOCTYPE KDjVuBookmarks>
+         * <item title="Title 1" destination="dest1">
+         *   <item title="Title 1.1" destination="dest1.1" />
+         *   ...
+         * </item>
+         * <item title="Title 2" destination="dest2">
+         * \endverbatim
+         */
+        const QDomDocument * documentBookmarks() const;
+
+        /**
+         * Reads the links and the annotations for the page \p pageNum
+         *
+         * For both \p links and \p annotations , you can pass either a valid pointer
+         * (in case you want to extract that kind of information), or a null pointer
+         * (if you don't want that information).
+         */
+        void linksAndAnnotationsForPage( int pageNum, QList<KDjVu::Link*> *links, QList<KDjVu::Annotation*> *annotations ) const;
+
+        /**
+         * Check if the image for the specified \p page with the specified
+         * \p width, \p height and \p rotation is already in cache, and returns
+         * it. If not, a null image is returned.
+         */
+        QImage image( int page, int width, int height, int rotation );
+
+        /**
+         * Export the currently open document as PostScript file \p fileName.
+         * \returns whether the exporting was successful
+          */
+        bool exportAsPostScript( const QString & fileName, const QList<int>& pageList ) const;
+
+        /**
+         * Export the currently open document as PostScript file.
+         * \returns whether the exporting was successful
+          */
+        bool exportAsPostScript( QFile* file, const QList<int>& pageList ) const;
+
+        /**
+         * Return the list of the text entities for the specified \p page, that matches the
+         * specified \p granularity.
+         */
+        QList<KDjVu::TextEntity> textEntities( int page, const QString & granularity ) const;
+
+        /**
+         * Enable or disable the internal rendered pages cache.
+         */
+        void setCacheEnabled( bool enable );
+        /**
+         * \returns whether the internal rendered pages cache is enabled
+         */
+        bool isCacheEnabled() const;
+
+        /**
+         * Return the page number of the page whose title is \p name.
+         */
+        int pageNumber( const QString & name ) const;
+
+        /**
+        * Return the page number of the page whose title is \p name.
+        */
+        int pageCount() const;
+    private:
+        class Private;
+        Private * const d;
+};
+
+#endif
```

Comment:
In the interest of keeping the codebase clean, I think this class can be heavily reduced, only a tiny fraction is actually used.

Reply:
I thought about it. I am not sure which functionality will be used in the future. So i decided left all functionality.

Because if we will want to extend djvu support, in KDjvu wrapper we will be able to find needed functionality.